src/tree.rs

```diff
@@ -294,3 +311,66 @@ impl<'a> TreeData for Tree<'a> {
         }
     }
 }
+
+impl Data {
+    pub fn render(&self, cfg: &Config) -> Text {
+        match &self.display {
+            Display::Highlight(tokens) => SyntaxToken::render(cfg, tokens),
+            Display::Raw(text) => Text::from(text.as_ref()),
+        }
+    }
+
+    fn raw(text: Cow<'static, str>) -> Self {
+        let lines: Vec<_> = text.lines().collect();
+        let long_line = lines.iter().max_by_key(|line| line.len());
+        let rows = lines.len();
+        let columns = long_line.map_or(0, |line| line.len());
+        Self {
+            display: Display::Raw(text),
+            rows,
+            columns,
+        }
+    }
+
+    fn highlight(tokens: Vec<SyntaxToken>) -> Self {
+        let (rows, columns) = SyntaxToken::get_size(&tokens);
+        Self {
+            display: Display::Highlight(tokens),
+            rows,
+            columns,
+        }
+    }
+
+    fn null(cfg: &Config) -> Self {
+        if cfg.disable_highlight {
+            Self::raw(Cow::Borrowed(""))
+        } else {
+            Self::highlight(vec![SyntaxToken::Null("null")])
+        }
+    }
+
+    fn string(cfg: &Config, s: String) -> Self {
+        if cfg.disable_highlight {
+            Self::raw(Cow::Owned(s))
+        } else {
+            Self::highlight(vec![SyntaxToken::String(s)])
+        }
+    }
+
+    fn number(cfg: &Config, num: String) -> Self {
+        if cfg.disable_highlight {
+            Self::raw(Cow::Owned(num))
+        } else {
+            Self::highlight(vec![SyntaxToken::Number(num)])
+        }
+    }
+
+    fn bool(cfg: &Config, b: bool) -> Self {
+        let b = if b { "true" } else { "false" };
+        if cfg.disable_highlight {
+            Self::raw(Cow::Borrowed(b))
+        } else {
```

Comment:
Is it a common thing in rust (gopher speaking) to use if else when an if + return would have been simpler to read ?

```
If (disabled highlighting) {
   Whatever
   return
}

// something
```

Reply:
Here, I'm returning the `if else` block as a variable, which is a common practice in Rust.

It's somewhat similar to the following approach:

```rust
let foo = if disable_highlighting { "true" } else { "false" };
```

This can't be done in Go, but it's a common code snippet in Rust, so I'm used to writing it this way.

Of course, the snippet you provided is also feasible and might be more familiar to non-Rust users.